gdal/frmts/jpegxr/jpegxrdataset.cpp

```diff
@@ -0,0 +1,1202 @@
+        case WMP_errNotYetImplemented: return "NotYetImplemented";
+        case WMP_errAbstractMethod: return "AbstractMethod:";
+        case WMP_errOutOfMemory: return "OutOfMemory";
+        case WMP_errFileIO: return "FileIO";
+        case WMP_errBufferOverflow: return "BufferOverflow";
+        case WMP_errInvalidParameter: return "InvalidParameter";
+        case WMP_errInvalidArgument: return "InvalidArgument";
+        case WMP_errUnsupportedFormat: return "UnsupportedFormat";
+        case WMP_errIncorrectCodecVersion: return "IncorrectCodecVersion";
+        case WMP_errIndexNotFound: return "IndexNotFound";
+        case WMP_errOutOfSequence: return "OutOfSequence:";
+        case WMP_errNotInitialized: return "NotInitialized";
+        case WMP_errMustBeMultipleOf16LinesUntilLastCall: return "MustBeMultipleOf16LinesUntilLastCall";
+        case WMP_errPlanarAlphaBandedEncRequiresTempFile: return "PlanarAlphaBandedEncRequiresTempFile";
+        case WMP_errAlphaModeCannotBeTranscoded: return "AlphaModeCannotBeTranscoded";
+        case WMP_errIncorrectCodecSubVersion: return "IncorrectCodecSubVersion";
+        default: return "Unknown";
+        }
+    }
+};
+
+/************************************************************************/
+/* ==================================================================== */
+/*                         JPEGXRDecoder                                */
+/* ==================================================================== */
+/************************************************************************/
+
+struct JPEGXRDecoder final
+{
+    JPEGXRDecoder()
+        : pCodecFactory(nullptr), pImageDecode(nullptr)
+    {
+    }
+
+    ~JPEGXRDecoder()
+    {
+        if (pImageDecode)
+            pImageDecode->Release(&pImageDecode);
+        pImageDecode = nullptr;
+
+        if (pCodecFactory)
+            pCodecFactory->Release(&pCodecFactory);
+        pCodecFactory = nullptr;
+    }
+
+    void Initialize(char const* pszFilename)
+    {
+        CPLAssert(!pCodecFactory);
+        CPLAssert(!pImageDecode);
+        ERR nErr = WMP_errFail;
+
+        /* -------------------------------------------------------------------- */
+        /*      Create decoder                                                  */
+        /* -------------------------------------------------------------------- */
+        nErr = PKCreateCodecFactory(&pCodecFactory, WMP_SDK_VERSION);
+        if (nErr != WMP_errSuccess || !pCodecFactory)
+            throw JPEGXRException(nErr);
+
+        nErr = pCodecFactory->CreateDecoderFromFile(pszFilename, &pImageDecode);
+        if (nErr != WMP_errSuccess || !pImageDecode)
+            throw JPEGXRException(nErr);
+
+        /* -------------------------------------------------------------------- */
+        /* Self-check understanding of jxrlib internals and data consistency.   */
+        /* -------------------------------------------------------------------- */
+#ifndef NDEBUG
```

Comment:
Interesting. From a quick look, it appears GDAL build systems never defines NDEBUG in release mode